x-pack/plugins/lens/public/editor_frame_service/embeddable/expression_wrapper.tsx

```diff
@@ -37,6 +39,55 @@ export interface ExpressionWrapperProps {
   className?: string;
 }
 
+interface VisualizationErrorProps {
+  errors: ExpressionWrapperProps['errors'];
+}
+
+export function VisualizationErrorPanel({ errors }: VisualizationErrorProps) {
+  const defaultMessage = (
+    <EuiFlexItem>
+      <EuiText size="s">
+        <FormattedMessage
+          id="xpack.lens.embeddable.failure"
+          defaultMessage="Visualization couldn't be displayed"
+        />
+      </EuiText>
+    </EuiFlexItem>
+  );
+  // only showing one error as for now
+  const customError =
+    errors == null ? null : (
+      <EuiFlexItem className="eui-textBreakAll">{errors[0].longMessage}</EuiFlexItem>
+    );
+  // show that there are more errors, but it requires to open the editor to actually see them
+  const moreErrors =
+    errors && errors.length > 1 ? (
+      <EuiFlexItem>
+        <FormattedMessage
+          id="xpack.lens.embeddable.moreErrors"
+          defaultMessage="(Edit in Lens editor to see more errors)"
+        />
+      </EuiFlexItem>
+    ) : null;
+  return (
+    <div className="lnsExpressionRenderer__embeddedError">
+      <EuiFlexGroup
+        direction="column"
+        justifyContent="center"
+        alignItems="center"
+        style={{ maxWidth: '100%' }}
+        data-test-subj="embeddable-lens-failure"
+      >
+        <EuiFlexItem grow={false}>
+          <EuiIcon type="alert" size="xl" color="danger" />
+        </EuiFlexItem>
+        {customError || defaultMessage}
+        {moreErrors}
+      </EuiFlexGroup>
+    </div>
```

Comment:
Is using an `EuiFlexGroup` as shown here the standard for displaying Dashboard error messages? I'd think it would be cleaner/easier to just use an `EuiEmptyPrompt` instead, since it already bakes in icon, text and action buttons. You could then also put your messages in a more semantic `p` elements, rather than the current `EuiFlexItem` wrappers. Thoughts?

Reply:
Nice idea. I used the same structure as in the Lens workspace but this one looks nicer.